client/widgets/project-health/health.directive.js

```diff
@@ -1,68 +1,186 @@
+  var avoidFetchByParams = true;
+  var paramsManuallyUpdated = false;
 
+  // Empty state chart to show when there is no data
+  var emptyStateChart = {
+    labels: [ '' ],
+    data: [ 1 ],
+    colors: [ '#7a7a7a' ]
   };
 
-  $scope.projectVersionChoices = [];
-
-  $scope.$watch('project', function(value) {
-    if (value) {
-      fetchMetrics();
-    }
+  // Set default configuration for the directive
+  _.defaults($scope, {
+    linkable: true,
+    compact: false,
+    filtersDisabled: false,
+    chartHeight: 200,
+    params: {
+      projectVersionId: null,
+      runnerId: null
+    },
+    stateChart: emptyStateChart,
+    showChart: false
   });
 
-  var ignoreChartParams = true;
-  $scope.$watch('chart.params', function(value) {
-    if (value && !ignoreChartParams) {
-      fetchMetrics();
-    }
+  if ($scope.project) {
+    fetchReportByTechnicalUser();
+  }
 
-    ignoreChartParams = false;
+  $scope.$watch('params', function () {
+    // Make sure to fetch report when params have been updated
+    if (!avoidFetchByParams && !paramsManuallyUpdated) {
+      fetchReport();
+    } else if (paramsManuallyUpdated) { // Reset the manual flag
+      paramsManuallyUpdated = false;
+    }
   }, true);
 
-  function fetchMetrics() {
-    var params = {};
+  /**
+   * Retrieve the latest report done by a technical user for the $scope.params. If not
+   * found, fallback to a standard lookup for a report
+   * @returns {*} A promise
+   */
+  function fetchReportByTechnicalUser() {
+    $scope.loading = true;
 
-    if ($scope.chart.params.projectVersionId) {
-      params.projectVersionId = $scope.chart.params.projectVersionId;
-    } else {
-      params.projectId = $scope.project.id;
+    // Do a call to the reports API to get the most recent report for a any technical user
+    return api({
+      url: '/reports',
+      params: buildReportsParams({ technical: true })
+    }).then(function(res) {
+      if (res.data.length == 1) {
+        // Retrieve the first technical runner
+        var technicalRunner = _.find(res.data[0].runners, function(runner) { return runner.technical; });
+
+        // Manually update the params to make sure the UI show the correct filter
+        paramsManuallyUpdated = true;
+        $scope.params.runnerId = technicalRunner.id;
+
+        return processData(res.data[0]);
+      } else {
+        return fetchReport();
+      }
+    });
+  }
+
+  /**
+   * Fetch a report by $scope.params and if there is no params, then it will
+   * try to retrieve the latest report of the project through project.lastReportId
+   * @returns {*} A promise
+   */
+  function fetchReport() {
+    $scope.loading = true;
+
+    // Use the reports API if there is any criteria
+    var url, params;
+    if ($scope.params.projectVersionId || $scope.params.runnerId) {
+      url = '/reports';
+      params = buildReportsParams();
+    } else if ($scope.project.lastReportId) { // Lookup by lastReportId
```

Comment:
This looks weird to me. Couldn't you simply update `buildReportParams` to always send the project ID, thus avoiding the need for this alternative request?

As a bonus, you would also no longer need `_.isArray(res.data) ? res.data[0] : res.data`